biopandas/mmcif/mmcif_parser.py

```diff
@@ -0,0 +1,444 @@
+# The CIF parser relieves heavily on the parser from PDBJapan (author: Gert-Jan Bekker).
+# Python CIF parser: https://gitlab.com/pdbjapan/tools/cif-parsers
+# License: MIT
+#   See https://gitlab.com/pdbjapan/tools/cif-parsers/blob/master/LICENSE
+import gzip
+import re
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    str.partition
+
+    def partitionString(string, sep):
+        return string.partition(sep)
+
+
+except:
+
+    def partitionString(string, sep):
+        tmp = string.split(sep)
+        return [tmp.pop(0), sep, sep.join(tmp)]
+
+
+class _loop:
+    def __init__(self, parserObj):
+        self.parserObj = parserObj
+        self.length = 0
+        self.refID = -1
+
+        self.refList = []
+
+        self.namesDefined = False
+
+    def addName(self, name):
+        catName = type(name) == str and partitionString(
```

Comment:
I am not quite sure what this is doing to be honest :). 

If `type(name) == str` is True and `partitionString(...)` is not None, then 

    catName = str

and otherwise

    catName = ["", "", ""]

is this intended?

Reply:
It's not a very intuitive line. The behaviour is:

```python
# name: str = "biopandas"
# catName = ('biopandas', '', '')

# name: str = "bio.pandas"
# catName = ('bio', '.', 'pandas')

# name: int = 12
# catName = ['', '', '']
```